components/device-mgt/org.wso2.carbon.device.mgt.core/src/main/java/org/wso2/carbon/device/mgt/core/service/DeviceOrganizationProviderServiceImpl.java

```diff
@@ -0,0 +1,331 @@
+package org.wso2.carbon.device.mgt.core.service;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.device.mgt.common.*;
+import org.wso2.carbon.device.mgt.core.dao.DeviceManagementDAOException;
+import org.wso2.carbon.device.mgt.core.dao.DeviceManagementDAOFactory;
+import org.wso2.carbon.device.mgt.core.dao.DeviceOrganizationDAO;
+import org.wso2.carbon.device.mgt.core.dao.DeviceOrganizationDAOException;
+import org.wso2.carbon.device.mgt.core.dao.impl.DeviceOrganizationDAOImpl;
+
+import java.util.*;
+
+public class DeviceOrganizationProviderServiceImpl implements DeviceOrganizationProviderService {
+
+    private static final Log log = LogFactory.getLog(DeviceOrganizationProviderServiceImpl.class);
+
+    DeviceOrganizationDAOImpl deviceOrganizationDAOimpl = new DeviceOrganizationDAOImpl();
+
+
+    @Override
+    public boolean addDeviceOrganization(String deviceId, String deviceName, String path, int pingMins, int state,
+                                         int isGateway) throws DeviceOrganizationException {
+        boolean isSuccess = false;
+        try {
+            DeviceManagementDAOFactory.beginTransaction();
+            isSuccess = deviceOrganizationDAOimpl.addDeviceOrganization(deviceId, deviceName, path, pingMins, state,
+                    isGateway);
+            DeviceManagementDAOFactory.commitTransaction();
+        } catch (DeviceOrganizationDAOException e) {
+            DeviceManagementDAOFactory.rollbackTransaction();
+            log.error(e.getMessage());
+            throw new DeviceOrganizationException(e);
+        } catch (TransactionManagementException e) {
+            String msg = "Error occurred while initiating transaction";
+            log.error(msg, e);
```

Comment:
IMO, we should throw this exception as well to the API layer in order to provide correct error descriptions to the client. Otherwise client might receive success response with "false" in payload though there is an internal server error. Please consider other places too.